packages/ui-kit/src/components/Button/Button.tsx

```diff
@@ -1,58 +1,166 @@
-import React from "react";
+import React, { ReactNode } from "react";
 import {
   StyleSheet,
-  TouchableOpacity,
   Text,
-  TouchableOpacityProps,
+  Image,
+  View,
+  Pressable,
+  PressableProps,
 } from "react-native";
 import { MergeProps } from "@/utils/types";
+import { usePressableState } from "../../utils/usePressableState";
+import { Icon, IconNames } from "../Icon/Icon";
 
 const buttonStyles = StyleSheet.create({
   base: {
     alignItems: "center",
-    justifyContent: "center",
     borderRadius: 100,
+    flexDirection: "row",
+    justifyContent: "center",
+  },
+  variant_solid: {
     backgroundColor: "#000",
   },
-  md: {
+  variant_solid_active: {
+    backgroundColor: "#353535",
+  },
+  variant_outline: {
+    backgroundColor: "#fff",
+    borderColor: "#000",
+  },
```

Comment:
Just FYI, these hardcoded colors are temporary, gonna move all colors to a constants file once we nail down colors a bit more.